src/emulator/io/include/io/file.h

```diff
@@ -0,0 +1,45 @@
+// Vita3K emulator project
+// Copyright (C) 2018 Vita3K team
+//
+// This program is free software; you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation; either version 2 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License along
+// with this program; if not, write to the Free Software Foundation, Inc.,
+// 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+
+#pragma once
+
+#include <psp2/types.h>
+#include <psp2/io/fcntl.h>
+#include <iostream>
+#include <cassert>
+#include <cstring>
+#include <vector>
+
+class ReadOnlyInMemFile {
+	std::vector<char> buf;
+	size_t            currentPos = 0;
```

Comment:
I think it's better to avoid manual alighment like that. Maybe run the format script (that runs clang-format) to see what are the stardard formatting rules.